tools/testing/test_selections.py

```diff
@@ -133,50 +137,121 @@ def _query_changed_test_files() -> List[str]:
     return lines
 
 
+def _get_previously_failing_tests() -> Set[str]:
+    PYTEST_FAILED_TESTS_CACHE_FILE_PATH = Path(".pytest_cache/v/cache/lastfailed")
+
+    if not PYTEST_FAILED_TESTS_CACHE_FILE_PATH.exists():
+        print(f"No pytorch cache found at {PYTEST_FAILED_TESTS_CACHE_FILE_PATH}")
+        print(f"cwd is {os.getcwd()}")
```

Comment:
Nit (also, please use `warnings.warn` to `print`, to make sure it is printed only once per process lifetime, rather than every time this method is called)
```suggestion
        print(f"No pytorch cache found at {PYTEST_FAILED_TESTS_CACHE_FILE_PATH.absolute()}")
```